lib/AST/SubstitutionMap.cpp

```diff
@@ -226,18 +226,14 @@ SubstitutionMap
 SubstitutionMap::getProtocolSubstitutions(ProtocolDecl *protocol,
                                           Type selfType,
                                           ProtocolConformanceRef conformance) {
-  auto protocolSelfType = protocol->getSelfInterfaceType();
-  return protocol->getGenericSignature()->getSubstitutionMap(
-    [&](SubstitutableType *type) -> Type {
-      assert(type->isEqual(protocolSelfType));
-      return selfType;
-    },
-    [&](CanType origType, Type replacementType, ProtocolType *protoType)
-      -> ProtocolConformanceRef {
-      assert(origType->isEqual(protocolSelfType) &&
-             protoType->getDecl() == protocol);
-      return conformance;
-    });
+  // The substitutions required for a protocol are simple: Self -> `selfType`,
+  // and Self: `protocol`.
+  auto protocolSelfType = cast<GenericTypeParamType>(
+      protocol->getSelfInterfaceType()->getCanonicalType());
+  SubstitutionMap map;
+  map.addSubstitution(protocolSelfType, selfType);
```

Comment:
Eventually SubstitutionMaps will be arrays indexed by generic signature requirements; how will this work?

Let's discuss this tomorrow.